### PR TITLE
Replace useDispatch with useQuery and request in PatientRegistration form.

### LINKED ISSUE #7171: Replace `useDispatch` w. `useQuery`/`request`: Patient Registration Form (`src/Components/Patient/PatientRegister.tsx`)

### PR DESCRIPTION
**Pull request by sriharsh05**
## Proposed Changes

- Fixes #7171 
- Replace useDispatch with useQuery and request in PatientRegister.tsx

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by sriharsh05** [comment]
Hey @rithviknishad ,

In the useQuery block of the `getPatient` API call, the patient details are not being assigned to the form. I have console logged the data, which is fetched at line no: 448. The data logs properly, but it is not being dispatched at line no: 449.

Can you please point out what mistake I made?

File: src/Components/Patient/PatientRegister.tsx [hunk 1]
```diff
@@ -225,6 +204,9 @@ export const PatientRegister = (props: PatientRegisterProps) => {
   );
   const [insuranceDetailsError, setInsuranceDetailsError] =
     useState<FieldError>();
+  const [stateId, setStateId] = useState<number>();
+  const [districtId, setDistrictId] = useState<number>();
+  const [localBodyId, setLocalBodyId] = useState<number>();
```
Comment (rithviknishad) [inline src/Components/Patient/PatientRegister.tsx]: Why are these states added? We are already tracking these in the form reducer right?

File: src/Components/Patient/PatientRegister.tsx [hunk 2]
```diff
@@ -297,97 +265,95 @@ export const PatientRegister = (props: PatientRegisterProps) => {
...
       });
 
       Promise.all([
-        fetchDistricts(res.data.district_object.state),
-        fetchLocalBody(res.data.district),
-        fetchWards(res.data.local_body),
-        duplicateCheck(res.data.mobile_number),
+        setStateId(data.district_object.state),
+        setDistrictId(data.district),
+        setLocalBodyId(data.local_body),
```
Comment (rithviknishad) [inline src/Components/Patient/PatientRegister.tsx]: What is a setState call doing inside Promise.all?

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Although the reducer state is being updated, the `field` from the `<Form>` is not being updated. `<Form>` already has a reducer to handle form state and errors. And in the Patient Register page we are keeping another reducer to handle form state and errors for the same purpose. This leads to not having a single source of truth.

Either use `<Form>` alone by removing the reducer and rely on it's `asyncDefaults` prop to pre-fill the form during edit form, or remove the usage of `<Form>` and rely on the custom defined reducer alone.

**Comment by rithviknishad** [comment]
Although not recommended, you could set key of the form with a value dependent on the form state (maybe a hash or something), to force update the Form's provider value. Use this as a last resort.

image

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the changes.

File: src/Components/Patient/PatientRegister.tsx (hunk 1 above)
Comment (rithviknishad) [reply src/Components/Patient/PatientRegister.tsx]: This is still unresolved.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the code.

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**